### PR TITLE
Day11-ex1-FeedActivity

### PR DESCRIPTION
**Pull request by at-dinhvo**
#### Description (Required)
- [x] Get data from database file
- [x] Show image user library universal

### CODE REVIEW

File: app/src/main/java/vn/asiantech/internship/drawer/ui/feed/FeedPagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        return imageLayout;
+    }
+
+    private void loadImage(String link, ImageView imageView) {
+        DisplayImageOptions defaultOptions = new DisplayImageOptions.Builder()
+                .cacheOnDisc(true).cacheInMemory(true)
+                .imageScaleType(ImageScaleType.EXACTLY)
+                .displayer(new FadeInBitmapDisplayer(300)).build();
+
+        ImageLoaderConfiguration config = new ImageLoaderConfiguration.Builder(
```
Comment (at-tamnguyen2) [inline app/src/main/java/vn/asiantech/internship/drawer/ui/feed/FeedPagerAdapter.java]: em chuyển toàn bộ code config vào activity hoặc constructor, hoặc application, không được config lúc load image như thế này. "Config tại một chỗ thôi"

**Comment by at-tamnguyen2** [comment]
@at-dinhvo em check code sửa lại arrow viewpager, nếu có 1 hình thì không hiện arrow nghe em. Thanks

File: app/src/main/java/vn/asiantech/internship/drawer/ui/feed/FeedPagerAdapter.java (hunk 1 above)
Comment (at-dinhvo) [inline app/src/main/java/vn/asiantech/internship/drawer/ui/feed/FeedPagerAdapter.java]: Da, ok anh :)

File: app/src/main/java/vn/asiantech/internship/drawer/ui/database/DatabaseHelper.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            mDatabase.close();
+        }
+        super.close();
+    }
+
+    public ArrayList<String[]> getAllData() {
+        ArrayList<String[]> links = new ArrayList<>();
+        String sql = "SELECT " + COL_LINK + " FROM " + TABLE_IMAGE;
+        Cursor cursor = mDatabase.rawQuery(sql, null);
+        String listUrl[];
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/drawer/ui/database/DatabaseHelper.java]: ERROR: Array brackets at illegal position.

File: app/src/main/java/vn/asiantech/internship/drawer/ui/database/DatabaseHelper.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    private String dataPath;
+    private Context mContext;
+
+    public DatabaseHelper(Context context) throws IOException {
+        super(context, DB_NAME, null, 1);
+        dataPath = context.getFilesDir().getPath();
+        this.mContext = context;
+        boolean dbexist = checkDatabase();
+        if (dbexist) {
+            System.out.println("Database exists");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/drawer/ui/database/DatabaseHelper.java]: ERROR: [SystemPrintln] System.out.println is used
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#SystemPrintln

File: app/src/main/java/vn/asiantech/internship/drawer/ui/database/DatabaseHelper.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            openDatabase();
+        } else {
+            System.out.println("Database doesn't exist");
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/drawer/ui/database/DatabaseHelper.java]: ERROR: [SystemPrintln] System.out.println is used
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#SystemPrintln

File: app/src/main/java/vn/asiantech/internship/drawer/ui/feed/FeedFragment.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        return layout;
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        try {
+            mDatabase = new DatabaseHelper(getContext());
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/drawer/ui/feed/FeedFragment.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

**Comment by at-tamnguyen2** [comment]
@at-dinhvo 
LGTM
Thank you

**Comment by at-tienhoang** [comment]
@at-dinhvo Please resolve conflict

File: app/src/main/res/values/dimens.xml [hunk 6]
```diff
@@ -4,12 +4,12 @@
     <dimen name="activity_vertical_margin">16dp</dimen>
     <dimen name="splash_item_margin">10dp</dimen>
     <!-- Bai5 -->
-    <dimen name="item.textview_padding">5dp</dimen>
-    <dimen name="item.textview_text_size">18sp</dimen>
-    <dimen name="item.cardview_padding">5dp</dimen>
-    <dimen name="item.textview_margin">10dp</dimen>
-    <dimen name="item.imageview_padding">5dp</dimen>
-    <dimen name="item.imageview_margin">10dp</dimen>
+    <dimen name="item.textView_padding">5dp</dimen>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.item_textView_padding` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -4,12 +4,12 @@
...
+    <dimen name="item.textView_text_size">18sp</dimen>
+    <dimen name="item.cardView_padding">5dp</dimen>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.item_cardView_padding` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/dimens.xml [hunk 8]
```diff
@@ -4,12 +4,12 @@
...
+    <dimen name="item.textView_margin">10dp</dimen>
+    <dimen name="item.imageView_padding">5dp</dimen>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.item_imageView_padding` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -4,12 +4,12 @@
...
+    <dimen name="item.imageView_margin">10dp</dimen>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.item_imageView_margin` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/dimens.xml [hunk 10]
```diff
@@ -22,20 +22,19 @@
     <dimen name="header.image_border_weight">3dp</dimen>
     <dimen name="item.layout_padding">5dp</dimen>
     <dimen name="item.layout_margin">10dp</dimen>
-    <dimen name="item.layout_minheight">60dp</dimen>
+    <dimen name="item.layout_minHeight">60dp</dimen>
     <dimen name="activity.toolbar_height">60dp</dimen>
     <dimen name="activity.toolbar_margin">20dp</dimen>
     <dimen name="activity.content_margin">20dp</dimen>
-    <dimen name="activity.leftmenu_width">240dp</dimen>
+    <dimen name="activity.leftMenu_width">240dp</dimen>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.activity_leftMenu_width` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/dimens.xml [hunk 11]
```diff
@@ -4,12 +4,10 @@
...
-    <dimen name="item.textview_padding">5dp</dimen>
-    <dimen name="item.textview_text_size">18sp</dimen>
-    <dimen name="item.cardview_padding">5dp</dimen>
-    <dimen name="item.textview_margin">10dp</dimen>
-    <dimen name="item.imageview_padding">5dp</dimen>
-    <dimen name="item.imageview_margin">10dp</dimen>
+    <dimen name="item.textView_padding">5dp</dimen>
+    <dimen name="item.textView_text_size">18sp</dimen>
+    <dimen name="item.textView_margin">10dp</dimen>
+    <dimen name="item.imageView_margin">10dp</dimen>
```
Comment (at-dev-support-01) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.item_imageView_margin` appears to be unused
 Unused resources make applications larger and slow down builds.

**Comment by at-tienhoang** [comment]
@at-dinhvo Please resolve conflict

**Pull request merged**